manifests/init.pp

```diff
@@ -591,6 +591,15 @@
   -> Class['::sensu::server::service']
   -> Class['::sensu::enterprise::dashboard']
 
+  # Dependencies for Plugins #
+  # Without this line, plugins installed with the sensu_gem provider may get
+  # evaluated before the sensu package, which, on sensu package upgrades that
+  # install a new version of Ruby, would mean that two runs of Puppet would be
+  # necessary to get the plugins installed into the new embedded Ruby folder.
+  Package['sensu']
+  ~> Package<| provider == 'sensu_gem' |>
+  ~> Service['sensu-client', 'sensu-server', 'sensu-api']
```

Comment:
Please test with the enterprise version as it uses different service names, we would need to take that into account.

Reply:
Good call, I'll test and most likely will ultimately add sensu-enterprise to the service list.  I think we're fine with only needing Package['sensu'] because I think the Sensu core package is the one that installs the embedded Ruby (not sure what Package['sensu-enterprise'] adds, but as long as it's something other than the Ruby implementation I don't care about it).

Reply:
Added in ebeeeef.....also lol at ebeeeef